src/components/ColorRampDropdown.tsx

```diff
@@ -6,20 +6,75 @@ import styles from "./ColorRampDropdown.module.css";
 import { DEFAULT_COLOR_RAMPS } from "../constants/color_ramps";
 import { AppThemeContext } from "./AppStyle";
 import IconButton from "./IconButton";
+import { DEFAULT_CATEGORICAL_PALETTES } from "../constants";
+import { Color } from "three";
 
 type ColorRampSelectorProps = {
-  selected: string;
-  onChange: (colorRampKey: string, reversed: boolean) => void;
+  selectedRamp: string;
+  onChangeRamp: (colorRampKey: string, reversed: boolean) => void;
   colorRamps?: typeof DEFAULT_COLOR_RAMPS;
+
+  useCategoricalPalettes?: boolean;
+  categoricalPalettes?: typeof DEFAULT_CATEGORICAL_PALETTES;
+  numCategories: number;
+  selectedPalette: Color[];
+  onChangePalette: (newPalette: Color[]) => void;
+
   disabled?: boolean;
   reversed?: boolean;
 };
 
 const defaultProps: Partial<ColorRampSelectorProps> = {
   colorRamps: DEFAULT_COLOR_RAMPS,
   disabled: false,
+  useCategoricalPalettes: false,
+  categoricalPalettes: DEFAULT_CATEGORICAL_PALETTES,
 };
 
+/**
+ * Returns a style object with a hard-stop linear gradient background, consisting
+ * of the colors in the provided array.
+ * @param colors Array of Color objects, in order, from left to right.
+ * @returns A style object with a linear gradient background.
+ *
+ * @example
+ * ```
+ * const colors = [new Color(0xffffff), new Color(0xcccccc), new Color(0x000000)];
+ * const style = getColorPaletteStyling(colors);
+ * // style = {background : linear-gradient(to right, #ffffff 0.00%, #ffffff 33.33%, #cccccc 33.33%, #cccccc 66.67%, #000000 66.67%, #000000 100.00%}
+ * ```
+ */
+function getColorPaletteStyling(colors: Color[]): { background: string } {
```

Comment:
Nit: Something about this function name seems odd but I'm not sure if `createCSSGradientForDiscretePalette` would be any better.